range.c

```diff
@@ -1268,10 +1286,15 @@ range_minmax(VALUE range)
     if (rb_block_given_p()) {
         return rb_call_super(0, NULL);
     }
-    return rb_assoc_new(
-        rb_funcall(range, id_min, 0),
-        rb_funcall(range, id_max, 0)
-    );
+    VALUE result_min = range_min_internal(0, NULL, range);
+    if (result_min == Qundef) {
+        result_min = rb_funcallv(range, id_min, 0, 0);
+    }
```

Comment:
I believe this code path cannot be reached.

For `range_min_internal` to return `Qundef`, `rb_block_given_p()` must be true. But `range_minmax` already checks that a few lines above and calls super `minmax`.